source/application/AMPIC887DriverServer/AMGCS2CommandFactory.cpp

```diff
@@ -0,0 +1,497 @@
+#include "AMGCS2CommandFactory.h"
+#include <QStringList>
+
+#include "AMGCS2Support.h"
+#include "GCS2Commands/AMGCS2MoveCommand.h"
+#include "GCS2Commands/AMGCS2GetCurrentPositionCommand.h"
+#include "GCS2Commands/AMGCS2StopCommand.h"
+#include "GCS2Commands/AMGCS2HaltSmoothlyCommand.h"
+#include "GCS2Commands/AMGCS2GetMovingStatusCommand.h"
+#include "GCS2Commands/AMGCS2CompositeCommand.h"
+#include "GCS2Commands/AMGCS2GetControllerReadyStatusCommand.h"
+#include "GCS2Commands/AMGCS2GetDeviceIdentificationCommand.h"
+#include "GCS2Commands/AMGCS2GetCommandLevelCommand.h"
+#include "GCS2Commands/AMGCS2SetCommandLevelCommand.h"
+#include "GCS2Commands/AMGCS2SetSyntaxVersionCommand.h"
+#include "GCS2Commands/AMGCS2GetDataRecorderConfigurationCommand.h"
+#include "AMPIC887DataRecorderConfiguration.h"
+#include "GCS2Commands/AMGCS2SetDataRecorderConfigurationCommand.h"
+#include "GCS2Commands/AMGCS2GetNumberOfRecordedPointsCommand.h"
+#include "GCS2Commands/AMGCS2GetRecordedDataValuesCommand.h"
+#include "GCS2Commands/AMGCS2SetDataRecorderConfigurationCommand.h"
+#include "GCS2Commands/AMGCS2SetRecordTriggerSourceCommand.h"
+#include "GCS2Commands/AMGCS2GetRecordTriggerSourceCommand.h"
+#include "GCS2Commands/AMGCS2ReferenceMoveCommand.h"
+#include "GCS2Commands/AMGCS2GetReferenceResultCommand.h"
+#include "GCS2Commands/AMGCS2GetRecorderOptionsCommand.h"
+#include "GCS2Commands/AMGCS2GetAvailableParametersCommand.h"
+#include "GCS2Commands/AMGCS2GetLimitSwitchStatusCommand.h"
+#include "GCS2Commands/AMGCS2GetTargetPositionCommand.h"
+#include "GCS2Commands/AMGCS2MoveRelativeCommand.h"
+#include "GCS2Commands/AMGCS2GetLowSoftLimitsCommand.h"
+#include "GCS2Commands/AMGCS2SetLowSoftLimitsCommand.h"
+#include "GCS2Commands/AMGCS2GetOnTargetStateCommand.h"
+#include "GCS2Commands/AMGCS2GetHighSoftLimitsCommand.h"
+#include "GCS2Commands/AMGCS2SetHighSoftLimitsCommand.h"
+#include "GCS2Commands/AMGCS2GetPositionUnitsCommand.h"
+#include "GCS2Commands/AMGCS2SetCycleTimeCommand.h"
+#include "GCS2Commands/AMGCS2GetCycleTimeCommand.h"
+#include "GCS2Commands/AMGCS2GetPivotPointCommand.h"
+#include "GCS2Commands/AMGCS2SetPivotPointCommand.h"
+#include "GCS2Commands/AMGCS2GetSoftLimitsStatusCommand.h"
+#include "GCS2Commands/AMGCS2SetSoftLimitsStatusCommand.h"
+#include "GCS2Commands/AMGCS2GetServoModeCommand.h"
+#include "GCS2Commands/AMGCS2SetServoModeCommand.h"
+#include "GCS2Commands/AMGCS2GetStepSizeCommand.h"
+#include "GCS2Commands/AMGCS2SetStepSizeCommand.h"
+#include "GCS2Commands/AMGCS2GetMinCommandablePositionCommand.h"
+#include "GCS2Commands/AMGCS2GetMaxCommandablePositionCommand.h"
+#include "GCS2Commands/AMGCS2GetSystemVelocityCommand.h"
+#include "GCS2Commands/AMGCS2SetSystemVelocityCommand.h"
+#include "GCS2Commands/AMGCS2VirtualMoveCommand.h"
+
+AMGCS2Command * AMGCS2CommandFactory::buildCommand(const QString &commandString)
+{
+	if(commandString.startsWith("POS?")) {
```

Comment:
Thanks chaps. 
I think what I might do is make a note of this, and get to it at a later date. It will certainly improve efficiency, but feel it can be something done down the road.

If nobody minds I'll create an issue for it, but not include it in the Hexapod milestone (as it isn't fundamental to its workings). This way I won't forget to get to it later.
@dretrex @davidChevrier Does this seem okay?